src/webui/api/torrentscontroller.cpp

```diff
@@ -1041,3 +1041,68 @@ void TorrentsController::categoriesAction()
 
     setResult(categories);
 }
+
+void TorrentsController::addTagsAction()
+{
+    checkParams({"hashes", "tags"});
+
+    const QStringList hashes {params()["hashes"].split('|')};
+    const QStringList tags {params()["tags"].split(',', QString::SkipEmptyParts)};
+
+    for (const QString &tag : tags) {
+        applyToTorrents(hashes, [&tag](BitTorrent::TorrentHandle *const torrent)
+        {
+            torrent->addTag(tag.trimmed());
```

Comment:
It's a bad idea to trim it in each iteration.

Reply:
Could you be more clear about it? Should we not trim or trim it in other places?